cmake/TaichiCore.cmake

```diff
@@ -3,6 +3,7 @@ option(TI_WITH_LLVM "Build with LLVM backends" ON)
 option(TI_WITH_METAL "Build with the Metal backend" ON)
 option(TI_WITH_CUDA "Build with the CUDA backend" ON)
 option(TI_WITH_CUDA_TOOLKIT "Build with the CUDA toolkit" OFF)
+option(TI_WITH_AMDGPU "Build with the AMDGPU backend" OFF)
```

Comment:
What if the user sets both `TI_WITH_CUDA=ON` and `TI_WITH_AMDGPU=ON`?

Reply:
It will compile both `CUDA` and `AMDGPU` devices. However, the exact device available at runtime depends on the GPU in the system.

Reply:
Given that only one of AMDGPU or CUDA will be available at runtime, I was thinking we should warn users not to enable both TI_WITH_CUDA and TI_WITH_AMDGPU at the same time. How about adding a check here to verify that?